### ISSUE TITLE
Error: Could not find or load main class * when executing neo4j

### ISSUE THREAD

**Issue by ellepannitto** [neo4j/neo4j#11965]
Neo4j version: don't know because I can't execute neo4j version
Operating system: Ubuntu 16.04 server
API/Driver: command line

**Steps to reproduce**
run neo4j version
Expected behavior
the version number is shown
Actual behavior
got this error: Could not find or load main class org.neo4j.server.CommunityEntryPoint

run neo4j-import [other parameters]
Expected behavior
the import process starts
Actual behavior
got this error: Could not find or load main class org.neo4j.tooling.ImportTool

run neo4j-admin import [other parameters]
Expected behavior
the import process starts
Actual behavior
Error: Could not find or load main class org.neo4j.commandline.admin.AdminTool

**Comment by craigtaverner** [comment]
Can I ask how you installed Neo4j? Normally when you install it you install a specific version (and get a download with the version number in the installation file), so in the normal case you would know the version. My suspicion here is that this was not installed in one of the normal ways, and perhaps as a result there are missing dependencies, or missing subdirectories, for example the contents of the lib directory where the classes should be that are not being found.

My advice would be to delete and re-install with a fresh download from https://neo4j.com/download-center/#releases.

**Comment by ian-cannon** [comment]
I am seeing the same issue.

Neo4j version: don't know because I can't execute neo4j version
Operating system: Ubuntu 18.04
API/Driver: command line
Neo4j version: neo4j 3.5.6

Installed neo4j using apt-get

**Comment by jennyowen** [comment]
@ian-cannon what does `java -version` return? Is it the same as `$JAVA_HOME`?
What command do you use to start neo4j?

**Comment by philastrophist** [comment]
I'm having the same problem. 

I've changed my conf file directories and the NEO4J_HOME and NEO4_CONF in order to have write permissions in the data directories but then `neo4j version` fails with the above error. 

If I remove my environment variables, `neo4j version` runs correctly but then I can't write to anything so it fails later due to permissions.

Catch 22 situation.
Neo4j appears to lose where the bin is or where java is (but the `which java` and `java -version` are the same in both cases)

How do I fix this?

**Comment by iNoSec2** [comment]
still not fixed??? if someone have find a fix, i will be happy to read it :)

**Comment by jennyowen** [comment]
@iNoSec2 no-one replied with the diagnostic information I asked for :disappointed: 

`Could not find or load main class ....` generally means the required library jars are not on the java classpath. It's not really a neo4j specific problem, more neo4j reacting badly with whatever java environment you have set up.

Maybe this means that you don't have java 11 as the default, or maybe you override the classpath somewhere in your environment setup?
The neo4j startup script should solve a lot of these environment problems (and it does for most people!). It could be then that you have a problem with that script? What does `which neo4j` return? Does `cat $(which neo4j)` return something that looks sensible?

**Comment by iNoSec2** [comment]
I will be on my PC later and will give you allbinfo you need ;)
I have tried multiple Java version from 8 to 16 xD and same exact result for all. Im on arch and i used yay to install it.
The systemctl start the service with no error and even a tool "bloodhound" can connect to the database, the error comes into play when i try to use the CLI, just neo4j version command doesnt work and show me the error i gave.
I tried the which neo4j already and cat it too and nothing shocking, it is in the normal /usr/bin folder and it is a bash script iirc.
Since Bloodhound work perfectly i dont really care about the error but maybe i will need it later so i want to investigate what is the problem :)
Let me know if you need some debug output/log.
Thanks for your answer

**Comment by jennyowen** [comment]
@iNoSec2 OK, you need java 11 for 4.0.0 and higher, and java 8 for 3.x.x. No need to try all the combinations :+1: 

Anyway, sorry but we don't support arch linux and we don't actually manage or produce the packages that are available on AUR. Those are done by someone unaffiliated with us. Someone else reported similar problems with the yay install: https://github.com/neo4j/neo4j/issues/12714
There's also been some debate about this start up problem on the package's arch linux page: https://aur.archlinux.org/packages/neo4j-community/

This all means that there's a good chance that the `yay` package is where your start-up problems are coming from. You'll need to either raise the issue with the package maintainer or try installing via tarball rather than `yay`.
Here's the instructions on tarball installation: https://neo4j.com/docs/operations-manual/current/installation/linux/tarball/

**Comment by iNoSec2** [comment]
Thanks a lot for your help

**Comment by giorgioditizio** [comment]
> still not fixed??? if someone have find a fix, i will be happy to read it :)

I was in a similar situation as @philastrophist. In my case, it seems it failed because by changing the NEO4J_HOME folder it cannot find the jar files in the Lib folder used by default by neo4j.
I solved by:
1)Creating the environmental variables NEO4J_HOME and NEO4J_CONF (both are needed) pointing respectively to the new home folder of neo4j and the folder containing the neo4j.conf file.
2)Modified the setting dbms.directories.lib in neo4j.conf file to point to the default Lib folder used by Neo4j  that contains all JAR files.

In alternative, one can avoid using the NEO4J_CONF environmental variable and add multiple PATHs to the NEO4J_HOME environmental variable. The first is a path to a directory I have write and read privileges, followed by the folder that contains the Lib folder created by neo4j during the installation. But I think it is better to work with the neo4j.conf file.

I checked and the following commands now work: neo4j version, neo4j start, neo4j console

**Comment by sowmiyaEE** [comment]
Yes that one helped @giorgioditizio Thanks
